utils/questGen.py

```diff
@@ -9,71 +17,165 @@ def generate_quest(quest):
+            'quest_type_raw': quest_type_raw, 'quest_type': quest_type_text, 'quest_condition': quest_condition, 'item_amount': item_amount, 'item_type': item_type, 
             'quest_target': quest_target, 'name': name, 'url': url, 'timestamp': timestamp, 'pokemon_id': pokemon_id, 'item_id': item_id,
-            'pokemon_name': pokemon_name, 'quest_reward_type': quest_reward_type, 'quest_reward_type_raw': quest_reward_type_raw})
+            'pokemon_name': pokemon_name, 'quest_reward_type': quest_reward_type, 'quest_reward_type_raw': quest_reward_type_raw, 'quest_task': quest_task})
         return quest_raw
     
 def questreward(quest_reward_type):
     type = {
-        2: "Item",
-        3: "Stardust",
-        7: "Pokemon"
+        2: _("Item"),
+        3: _("Stardust"),
+        7: _("Pokemon")
     }
     return type.get(quest_reward_type, "nothing")
     
 def questtype(quest_type):
-    with open('utils/quest/types.json') as f:
-        items = json.load(f)
-    return (items[str(quest_type)]['text'])
+    file = open_json_file('types')
+    return (file[str(quest_type)]['text'])
     
 def rewarditem(itemid):
-    with open('utils/quest/items.json') as f:
-        items = json.load(f)
-    return (items[str(itemid)]['name'])
+    file = open_json_file('items')
+    return (file[str(itemid)]['name'])
     
 def pokemonname(id):
-    with open('pokemon.json') as f:
-        mondata = json.load(f)
-    return mondata[str(int(id))]["name"]
+    file = open_json_file('pokemon')
+    return file[str(int(id))]["name"]
+
+def questtask(typeid, condition, target):
+    pokemonTypes = open_json_file('pokemonTypes')
+    items = open_json_file('items')
+    throwTypes = {"10":_("Nice"), "11":_("Great"), "12":_("Excellent"), "13":_("Curveball")}
+    arr = {}
+    arr['0'] = target
+    text = questtype(typeid)
+
+    if typeid == 4:
+        arr['wb'] = ""
+        arr['type'] = ""
+        text = _("Catch {0} {type}Pokemon{wb}.")
+        match_object = re.search(r"'pokemon_type': \[([0-9, ]+)\]", condition)
+        if match_object is not None:
+                pt = match_object.group(1).split(', ')
+                last = len(pt)
+                cur = 1
+                if last == 1:
+                    arr['type'] = pokemonTypes[pt[0]].title() + _('-type ')
+                else:
+                    for ty in pt:
+                        arr['type'] += (_('or ') if last == cur else '') + pokemonTypes[ty].title() + ('-type ' if last == cur else '-, ')
+                        cur += 1
+        if re.search(r"'type': 3", condition) is not None:
+                arr['wb'] = _(" with weather boost")
+        match_object = re.search(r"'pokemon_ids': \[([0-9, ]+)\]", condition)
+        if match_object is not None:
+                pt = match_object.group(1).split(', ')
+                last = len(pt)
+                cur = 1
+                if last == 1:
+                    arr['poke'] = i8ln(pokemonname[pt[0]])
```

Comment:
@evilmoses
Do you potentially need this too in case there are multiple pokemon?

else: for ty in pt: arr['poke'] += ('or ' if last == cur else '') + pokemonname[ty] cur += 1